src/components/TemplateContent.vue

```diff
@@ -0,0 +1,221 @@
+        <img v-else-if="this.status === 'accepted'" :src="checkSentSvg">
+        <img v-else-if="this.status === 'received'" :src="doubleCheckReceivedSvg">
+        <img v-else-if="this.status === 'consumed'" :src="doubleCheckReadSvg">
+        <div v-else-if="this.status === 'failed'" class="failure">
+          {{ failedToSendMsg }}
+        </div>
+      </span>
+      <div>{{ date }}</div>
+    </div>
+  </div>
+</template>
+
+<script>
+
+import { default as base } from '../mixins/baseComponent.js'
+import alertSvg from '../assets/img/alert.svg'
+import alertWhiteSvg from '../assets/img/alertWhite.svg'
+import paperplaneSvg from '../assets/img/paperplane.svg'
+import { formatText } from '@/utils/FormatTextUtils'
+
+export default {
+  name: 'template-content',
+  mixins: [
+    base
+  ],
+  props: {
+    document: {},
+    status: {
+      type: String,
+      default: ''
+    },
+    unsupportedContentMsg: {
+      type: String,
+      default: 'Unsuported Content'
+    },
+    contentTemplateName: {
+      type: String,
+      default: 'Unsuported Content'
+    },
+    failedToSendMsg: {
+      type: String,
+      default: 'Falha ao enviar a mensagem'
+    },
+    onFailedClickIcon: {
+      type: Function
+    }
+  },
+  data: function () {
+    return {
+      alertSvg,
+      alertWhiteSvg,
+      paperplaneSvg
+    }
+  },
+  computed: {
+    showTemplateContentTitle() {
+      return this.unsupportedContentMsg.replace(/_/g, ' ')
+    }
+  },
+  methods: {
+    showTemplateContentBody() {
+      let componentTemplateBody = this.document.templateContent.components
+        .filter(m => m.type === 'BODY')
+
+      let bodyFilledVariables = this.document.template.components
+        .filter(m => m.type === 'body')
+
+      if (componentTemplateBody) {
+        componentTemplateBody = componentTemplateBody[0].text
+
+        if (bodyFilledVariables.length) {
+          bodyFilledVariables = bodyFilledVariables[0].parameters
+          const regexFindVariablesInText = /\{\{(\d+)\}\}/g
+          componentTemplateBody = componentTemplateBody.replace(regexFindVariablesInText, (match, index) => {
+            const value = bodyFilledVariables[index - 1].text
+            return value !== undefined ? value : match
+          })
+        }
+      }
+
+      return componentTemplateBody
+    },
+    showTemplateContentLinks() {
+      const attachments = this.document.template.components
+        .filter((m) => (m.type === 'header'))
+
+      if (attachments) {
+        const links = []
+        attachments.forEach(item => {
+          item.parameters.forEach(parameter => {
+            if (parameter.type === 'image') {
+              links.push(parameter.image.link)
+            } else if (parameter.type === 'video') {
+              links.push(parameter.video.link)
+            } else if (parameter.type === 'audio') {
+              links.push(parameter.audio.link)
+            }
```

Comment:
```suggestion
           const { type } = parameter
            if (['image', 'video', 'audio'].includes(type)) {
              links.push(parameter[type].link)
            } 
```

I don't know if you can remove that second `forEach` but I think you can simplify this `if` like this.  What do you think?

Reply:
Yes, I agree on the simplification, I did not know something like this was possible in JS, thanks for the suggestion.

And about forEach, I tried to remove it but the implicit logic in the object blocks this improvement